hydra-main/src/main/java/com/addthis/hydra/query/aggregate/QueryTask.java

```diff
@@ -136,6 +150,29 @@ private void rescheduleSelfWithBackoff() {
         }
     }
 
+    private FrameReadTaskSourceProvider createTaskSourceProvider(MeshSourceAggregator sourceAggregator) {
+        Query query = sourceAggregator.query;
+        int totalTasks = sourceAggregator.totalTasks;
+        int maxSimul = getMaxSimul(query.getParameter("maxSimul"), totalTasks);
+        if (maxSimul == totalTasks) {
+            return new DefaultFrameReadTaskSourceProvider(sourceAggregator);
+        } else {
+            return new MaxSimulFrameReadTaskSourceProvider(sourceAggregator, maxSimul);
+        }
+    }
+
+    private int getMaxSimul(@Nullable String maxSimulQueryParam, int totalTasks) {
+        if (maxSimulQueryParam == null) {
+            return totalTasks;
+        }
+        try {
```

Comment:
Throwing IAE there will cause 500 response code, which is not any better. The correct solution is to validate the parameter in HttpQueryCallHandler and return 400. There are many unvalidated parameters with the same behavior. Feel free to open a ticket to improve our API overall. I consider that out of scope of this ticket though.